src/basilisp/contrib/nrepl_server.lpy

```diff
@@ -0,0 +1,479 @@
+;; adapted from
+;; https://github.com/babashka/nbb/blob/bca8b5017a06768eb35d02a2d6233ca9c6c2f692/src/nbb/impl/nrepl_server.cljs
+(ns basilisp.contrib.nrepl-server
+  "A port of `nbb <https://github.com/babashka/nbb>`_'s nREPL server implementation to basilisp."
+  (:require [basilisp.contrib.bencode :as bc]
+            [basilisp.string :as str])
+  (:import logging
+           [basilisp.lang.runtime :as rt]
+           socketserver
+           sys
+           traceback
+           uuid))
+
+(def ^:private logger
+  "The logger for this namespace."
+  (logging/getLogger (namespace ::)))
+
+(defmacro ^:private debug [& values]
+  (when (.isEnabledFor logger logging/DEBUG)
+    `(.debug logger (str/join " " [~@values]))))
+(defmacro ^:private info [& values]
+  (when (.isEnabledFor logger logging/INFO)
+    `(.info logger (str/join " " [~@values]))))
+(defmacro ^:private warn [& values]
+  (when (.isEnabledFor logger logging/WARNING))
+  `(.warning logger (str/join " " [~@values])))
+(defmacro ^:private error [& values]
+  ;; we want error to always output to `sys/stderr` unfiltered.
+  `(.write sys/stderr (str ~(keyword (str *ns*) "error") " " (str/join " " [~@values]) \newline)))
+
+(definterface ^:private IReadWrite
+  ;; Pythonic interface for creating `sys/stdout` like File objects.
+  (read [])
+  (write [value]))
+
+(deftype ^:private StreamOutFn [out-fn]
+  ;; A type to use as replacement binding for writing to `sys/stdout`
+  ;; stream, so that the output ``value`` is passed to ``out-fn``
+  ;; instead.
+  IReadWrite
+  (read [_this]
+    (throw (python/RuntimeError (str ::logic-error " shouldn't be possible to get here..."))))
+  (write [_self value]
+    (out-fn value)))
+
+(defn- response-for-mw [handler]
+  (fn [{:keys [id session] :as request} response]
+    (let [response (cond-> (assoc response
+                                  "id" id)
+                     session (assoc "session" session))]
+      (handler request response))))
+
+(defn- coerce-request-mw [handler]
+  (fn [request send-fn]
+    (handler (update request :op keyword) send-fn)))
+
+(defn- log-request-mw [handler]
+  (fn [request send-fn]
+    (info :request (dissoc request :client*))
+    (handler request send-fn)))
+
+(defn- log-response-mw [handler]
+  (fn [request response]
+    (info :response response)
+    (handler request response)))
+
+(declare ops)
+
+(defn- handle-describe [request send-fn]
+  (send-fn request
+           {"versions" {"basilisp" (let [version rt/BASILISP-VERSION-STRING]
```

Comment:
```suggestion
           {"versions" {"basilisp" (let [version basilisp.lang.runtime/BASILISP-VERSION-STRING]
```

This will compile down to a direct reference. No import is required because `basilisp.lang.runtime` is available in all modules.

Reply:
Updated, there was a second ref as well.